lib/Core/Executor.cpp

```diff
@@ -276,38 +250,115 @@ namespace {
-      "max-stack-frames",
-      cl::desc("Terminates a state after the limit of stack frames is reached "
-               "(default=8192). Disable check with 0."),
-      cl::init(8192));
+                   cl::desc("Inhibit forking at memory cap (vs. random terminate) (default=on)"),
+                   cl::init(true),
+                   cl::cat(TerminationCat));
+
+  cl::opt<unsigned> RuntimeMaxStackFrames("max-stack-frames",
+                                          cl::desc("Terminate a state after this many stack frames (default=8192).  Disable check with 0."),
+                                          cl::init(8192),
+                                          cl::cat(TerminationCat));
+
+  cl::opt<std::string>
+  MaxInstructionTime("max-instruction-time",
+                     cl::desc("Allow a single instruction to take only this much time (default=0s (off)). Enables --use-forked-solver"),
+                     cl::cat(TerminationCat));
+
+  
+
+  /*** Debugging options ***/
+
+  /// The different query logging solvers that can switched on/off
+  enum PrintDebugInstructionsType {
+    STDERR_ALL, ///
+    STDERR_SRC,
+    STDERR_COMPACT,
+    FILE_ALL,    ///
+    FILE_SRC,    ///
+    FILE_COMPACT ///
+  };
+
+  llvm::cl::bits<PrintDebugInstructionsType> DebugPrintInstructions(
+      "debug-print-instructions",
+      llvm::cl::desc("Log instructions during execution."),
+      llvm::cl::values(
+          clEnumValN(STDERR_ALL, "all:stderr", "Log all instructions to stderr "
+                                               "in format [src, inst_id, "
+                                               "llvm_inst]"),
+          clEnumValN(STDERR_SRC, "src:stderr",
+                     "Log all instructions to stderr in format [src, inst_id]"),
+          clEnumValN(STDERR_COMPACT, "compact:stderr",
+                     "Log all instructions to stderr in format [inst_id]"),
+          clEnumValN(FILE_ALL, "all:file", "Log all instructions to file "
+                                           "instructions.txt in format [src, "
+                                           "inst_id, llvm_inst]"),
+          clEnumValN(FILE_SRC, "src:file", "Log all instructions to file "
+                                           "instructions.txt in format [src, "
+                                           "inst_id]"),
+          clEnumValN(FILE_COMPACT, "compact:file",
+                     "Log all instructions to file instructions.txt in format "
+                     "[inst_id]")
+          KLEE_LLVM_CL_VAL_END),
+      llvm::cl::CommaSeparated);
+#ifdef HAVE_ZLIB_H
+  cl::opt<bool> DebugCompressInstructions(
+      "debug-compress-instructions", cl::init(false),
+      cl::desc("Compress the logged instructions in gzip format."));
+#endif
+
+  cl::opt<bool>
+  DebugCheckForImpliedValues("debug-check-for-implied-values");
+
+
+  cl::opt<double>
+  MaxStaticForkPct("max-static-fork-pct", 
+		   cl::init(1.),
+		   cl::desc("(default=1.0)"));
```

Comment:
```suggestion
      cl::desc("Set the maximum threshold for which at an instruction can be "
               "forked over the forks of all instructions. (default=1.0 "
               "(always)). Otherwise concretize."));
```